### PR TITLE
Reduce verbosity when reporting Excel cell formatting

### PR DESCRIPTION
**Pull request by CyrilleB79**
### Link to issue number:
None
### Summary of the issue:
When requesting  for cell formatting information in Excel (NVDA+F), there is over-verbose information:
* "no border lines" is often reported, since Excel files with default borders are quite common
* "background pattern none" is almost always reported since cells with other background patterns are very rare

### Description of user facing changes
When formatting information is requested with NVDA+F, we will not report anymore a lack of specific formatting for border lines or for background pattern. This formatting will still be reported during navigation if the corresponding option is enabled in document formatting settings (respectively "cell borders" and "color").

In that sense, we follow the logic used for other formatting attributes that have a negative state (lack of such formatting) such as bold/italic/etc. E.g.: "no bold" is not formatted upon NVDA+F request, but it is still reported while navigating from bold to no bold if the corresponding doc formatting option is enabled.

### Description of development approach
Do not provide info on background pattern in formatting info fields and changed the logic in `speech.py` to handle this.

For UIA, nothing is done since NVDA does not report borders or background pattern for Excel cells when UIA is enabled.

### Additional note
`_getFormatFieldAndOffsets` is private but it's probably called by `getTextWithFields`, thus changing its result.
Thus should I indicate something in API breaking changes though?

### Testing strategy:
Manual test: Tested NVDA+F and navigation with option "color" and "cell borders" enabled.
### Known issues with pull request:
None

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

### CODE REVIEW

**Comment by lukaszgo1** [comment]
> Excel UIA not supported.

Have you tried to implement the same fix for UIA and encountered some difficulties, or is that because when accessing Excel via UIA this problem does not occur?

**Comment by CyrilleB79** [comment]
> > Excel UIA not supported.
> 
> Have you tried to implement the same fix for UIA and encountered some difficulties, or is that because when accessing Excel via UIA this problem does not occur?

The problem of such verbosity is not present with UIA. But with UIA, there seem to be an error with this build when requesting formatting info; this error does not occur in alpha build.

Converting to draft to clarify this.

**Comment by CyrilleB79** [comment]
> The problem of such verbosity is not present with UIA.

Initial description updated now.

> But with UIA, there seem to be an error with this build when requesting formatting info; this error does not occur in alpha build.

There was actually no error; tests probably not executed correctly the first time.
So I'am setting this PR as ready again.

File: source/speech/speech.py [hunk 1]
```diff
@@ -2422,7 +2422,11 @@ def getFormatFieldSpeech(  # noqa: C901
 			textList.append(_("{backgroundColor} background").format(backgroundColor=bgColorText))
 		backgroundPattern=attrs.get("background-pattern")
 		oldBackgroundPattern=attrsCache.get("background-pattern") if attrsCache is not None else None
-		if backgroundPattern and backgroundPattern!=oldBackgroundPattern:
+		if (backgroundPattern or oldBackgroundPattern is not None)and backgroundPattern != oldBackgroundPattern:
```
Comment (seanbudd) [inline source/speech/speech.py:2425]:
```suggestion
		if (backgroundPattern or oldBackgroundPattern is not None) and backgroundPattern != oldBackgroundPattern:
```

**Review by seanbudd — APPROVED** [review]
Thanks @CyrilleB79

File: source/speech/speech.py [hunk 2]
```diff
@@ -2422,7 +2422,7 @@ def getFormatFieldSpeech(  # noqa: C901
 			textList.append(_("{backgroundColor} background").format(backgroundColor=bgColorText))
 		backgroundPattern=attrs.get("background-pattern")
 		oldBackgroundPattern=attrsCache.get("background-pattern") if attrsCache is not None else None
-		if (backgroundPattern or oldBackgroundPattern is not None)and backgroundPattern != oldBackgroundPattern:
+		if (backgroundPattern or oldBackgroundPattern is not None) and backgroundPattern != oldBackgroundPattern:
```
Comment (CyrilleB79) [inline source/speech/speech.py:2425]: Thanks for the linting. But why hasn't it been detected by appVeyor check?

Comment (seanbudd) [reply source/speech/speech.py:2425]: I assume a flake8 bug

**Pull request merged**